src/CashuWallet.ts

```diff
@@ -662,11 +662,25 @@ class CashuWallet {
 	 * @param invoice LN invoice that needs to get a fee estimate
 	 * @returns the mint will create and return a melt quote for the invoice with an amount and fee reserve
 	 */
-	async createMeltQuote(invoice: string): Promise<MeltQuoteResponse> {
+	async createMeltQuote(invoice: string, amount?: number): Promise<MeltQuoteResponse> {
 		const meltQuotePayload: MeltQuotePayload = {
 			unit: this._unit,
 			request: invoice
 		};
+
+		if (this._mintInfo?.amountless) {
```

Comment:
We should only be adding the amount, if the caller explicitly passes amount. Otherwise this implementation defaults to 0 and mints that support this will throw an error if amount != invoiceAmount.

Also amount passed needs to be in msats